dockstore-client/src/main/java/io/dockstore/client/cli/YamlClient.java

```diff
@@ -0,0 +1,92 @@
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package io.dockstore.client.cli;
+
+import java.util.List;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.ParameterException;
+import com.beust.jcommander.Parameters;
+import io.dockstore.client.cli.YamlVerifyUtility.ValidateYamlException;
+
+import static io.dockstore.client.cli.ArgumentUtility.out;
+import static io.dockstore.client.cli.JCommanderUtility.printJCommanderHelp;
+import static io.dockstore.client.cli.YamlVerifyUtility.DOCKSTOREYML;
+import static io.dockstore.client.cli.YamlVerifyUtility.YAML;
+
+public final class YamlClient {
+
+    public static final String VALIDATE_HELP_MESSAGE = "Verifies " + DOCKSTOREYML + " is correct and ensures all required files are present";
+
+    private YamlClient() {
+        // disable constructor for utility class
+    }
+
+
+    /**
+     * @param args
+     */
+    public static boolean handleCommand(List<String> args) {
+        String[] argv = args.toArray(new String[args.size()]);
+        JCommander jc = new JCommander();
+
+        CommandYaml commandYaml = new CommandYaml();
+        JCommander jcPlugin = JCommanderUtility.addCommand(jc, YAML, commandYaml);
+        CommandYamlValidate commandYamlValidate = new CommandYamlValidate();
+        JCommanderUtility.addCommand(jcPlugin, "validate", commandYamlValidate);
+        try {
+            jcPlugin.parse(argv);
+        } catch (ParameterException ex) {
+            printJCommanderHelp(jc, "dockstore", YAML);
+            out(ex.getMessage());
+            return true;
+        }
+        if (commandYaml.help) {
+            printJCommanderHelp(jc, "dockstore", YAML);
+        } else {
+            switch (jcPlugin.getParsedCommand()) {
+            case "validate":
+                try {
+                    YamlVerifyUtility.dockstoreValidate(CommandYaml.path);
+                } catch (ValidateYamlException ex) {
+                    out(ex.getMessage());
+                }
+                break;
+            default:
+                // fall through
+            }
+        }
+        return true;
+    }
+
+
+    @Parameters(separators = "=", commandDescription = "Tools used for " + YAML + " files")
+    private static class CommandYaml {
+        @Parameter(names = "--path", description = "Path to " + DOCKSTOREYML + " (ex. /home/usr/Dockstore/test, ~/Dockstore/test, or ../test)", required = true)
+        private static String path = null;
```

Comment:
The reason I used statics here is because `YamlClient.handleCommand` is declared as static and if `CommandYaml` isn't declared as static I run into errors. It appears this was done in [PluginClient](https://github.com/dockstore/dockstore-cli/blob/ac2ac87d1c738376e981b83cbf17d8b9c4b3241e/dockstore-client/src/main/java/io/dockstore/client/cli/PluginClient.java#L131).

Reply:
It's solvable by having a reference to the Command instance, something like [here](https://github.com/dockstore/dockstore-deploy/blob/f9626a876112559559fc869faabe6985676699f8/deploy-cli/src/main/java/io/dockstore/deploy/cli/Main.java#L93). But it can stay the way it is, although see next paragraph.

I just noticed, the way you run this is:

```
dockstore yaml --path <dir> validate
```

Where as I would expect the pattern to be:

```
dockstore yaml validate --path <dir>
```

That's how other commands work:

```
dockstore workflow launch --entry <entry>

### NOT
dockstore workflow --entry <entry> launch 
```

Which means, ignoring the whole static issue, that the property maybe belongs in `CommandYamlValidate`.